forum/qa-plugin/report-reason/q2apro-flag-reasons-page.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+class q2apro_flag_reasons_page
+{
+    
+    private $directory;
+    private $urlToRoot;
+    
+    public function load_module($directory, $urlToRoot)
+    {
+        $this->directory = $directory;
+        $this->urlToRoot = $urlToRoot;
+    }
+
+    public function suggest_requests() 
+    {    
+        return [
+            [
+                'title' => 'Ajax Flagger', // title of page
+                'request' => 'ajaxflagger', // request name
+                'nav' => 'M', // 'M'=main, 'F'=footer, 'B'=before main, 'O'=opposite main, null=none
+            ],
+        ];
+    }
+
+    public function match_request($request)
+    {
+        if ($request === 'ajaxflagger') {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function process_request($request)
+    {    
+        /*** 
+        Incoming reasonid for flags:
+        1 - spam
+        2 - quality
+        3 - rude
+        4 - edit
+        5 - migrate
+        6 - other
+        */
+
+        if(!qa_is_logged_in()) {
+            exit();
+        }
+
+        $transferString=$_POST['ajaxdata'];
```

Comment:
niefiltrowany w żaden sposób $_POST - potencjalne miejsce ataku

Reply:
htmlspecialchars?

Reply:
Dobra, zmieniłem na funkcje z Q2A, mam nadzieje, że gdzieś to filtruje.